pandas/tests/arrays/test_array.py

```diff
@@ -28,6 +29,20 @@
 )
 
 
+def test_dt64_array():
+    # PR 53439
+    dtype_unit_lst = ["M8[h]", "M8[m]", "m8[h]", "M8[m]"]
+
+    for unit in dtype_unit_lst:
+        dtype_var = np.dtype(unit)
+        msg = (
+            f"dtype={dtype_var} is not supported. Supported resolutions are 's', 'ms', "
+            "'us', and 'ns'"
+        )
+        with pytest.raises(TypeError, match=re.escape(msg)):
```

Comment:
this needs to issue a FutureWarning saying it will raise in the future (or cast to closest-supported-unit), not raise